pkg/federation/get.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2023 The Bestchains Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/bestchains/bc-cli/pkg/common"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/cmd/get"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+func NewFedGetCmd(option common.Options) *cobra.Command {
+	var (
+		labelSelector string
+		fieldSelector string
+	)
+
+	defaultPrintFlag := get.NewGetPrintFlags()
+	cmd := &cobra.Command{
+		Use: "fed [ORG-NAME] [FED-NAME]",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// get config from getter
+			cfg, err := clientcmd.BuildConfigFromKubeconfigGetter("", common.InKubeGetter)
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			cli, err := dynamic.NewForConfig(cfg)
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			list := corev1.List{
+				TypeMeta: v1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "v1",
+				},
+				ListMeta: v1.ListMeta{},
+			}
+
+			var obj runtime.Object
+
+			//
+			if len(args) == 0 {
+				feds, err := cli.Resource(schema.GroupVersionResource{Group: common.IBPGroup, Version: common.IBPVersion, Resource: common.FederationResource}).List(context.TODO(), v1.ListOptions{
```

Comment:
`List` seems not working for `Federations`. Have you tried this command before make this PR?

Reply:
Users only have get access to the federation, not list access. So we need to get a certain organization, get the feds the organization has joined, and get the list of feds.